main/zoe/api/zoe-helpers.md

```diff
@@ -300,3 +221,91 @@ const offerHook = offerHandle => {
     });
 };
 ```
+
+## zoeHelpers.satisfies()
+- `offerhandle`- The offer being checked
+- `allocation` - The allocation checked against the offer's wants.
+- Returns: `true` if the allocation satisfies the offer, `false` if not.
+
+Checks is an allocation would satisfy a single offer's wants if that was the allocation passed to
+`reallocate()`. This is half of the offer safety check; whether the allocation constitutes a refund
+is not checked. 
+```js
+//If `leftOfferHandle' is:
+//{ give: { Asset: moola(10) },
+//  want: { Price: simoleans(4) },
+// giving someone exactly what they want satisifies wants, returns `true`
+satisfies(leftOfferHandle, {
+  Asset: moola(0),
+  Price: simoleans(4),
+})
+// giving someone less than what they want even with a refund doesn't satisfy wants`,
+satisfies(leftOfferHandle, {
+  Asset: moola(10),
+  Price: simoleans(3),
+})
+//giving someone less than what they want even with a refund doesn't satisfy wants`,      
+satisfies(leftOfferHandle, {
+  Asset: moola(0),
+  Price: simoleans(3),
+}),
+```
+
+## zoeHelpers.isOfferSafe()
+- `offerHandle` - Offer being checked.
+- `allocation` - Proposed allocation
+- Returns: `true` if the allocation is safe for the offer, `false` otherwise.
+checks whether an
+allocation for a particular offer would satisfy offer safety. Any
+allocation that returns true under `satisfy` **(tyg todo: Should this be "satisfies"?)** will also return true
+under `isOfferSafe`. (`isOfferSafe` is equivalent of `satisfies` || **(tyg todo: I'd prefer to use "logical OR" here. OK?)**
+gives a refund).
+```js
+//If `leftOfferHandle` is:
+//{ give: { Asset: moola(10) },
+//  want: { Price: simoleans(4) },
+//Giving someone exactly what they want is offer safe
+isOfferSafe(leftOfferHandle, {
+  Asset: moola(0),
+  Price: simoleans(4),
+})
+// Giving someone less than what they want and not what they gave is not offer safe
+isOfferSafe(leftOfferHandle, {
+  Asset: moola(0),
+  Price: simoleans(3),
+})  ,
+```
+
+## zoeHelpers.trade()
+- **(tyg todo: Not sure how to describe/name the arguments with respect to their being a combination of offerHandle, gain, and loss?)**
```

Comment:
Yeah. It's a little tricky. Trade takes two arguments. Each argument is a record with keys: `offerHandle`, `gains`, and `losses` (losses is optional). The value of offerHandle is an offerHandle. `gains` and `losses` are amountKeywordRecords describing declaratively what is added or removed from the allocation for that offer.